structures/entitlement.go

```diff
@@ -43,10 +42,12 @@ func (b EntitlementBuilder) GetUser(ctx context.Context, inst mongo.Instance) (*
 		return nil, fmt.Errorf("Entitlement does not have a user assigned")
 	}
 
-	ub, err := UserBuilder{}.FetchByID(ctx, inst, b.Entitlement.UserID)
-	if err != nil {
+	var user *User
+	if err := inst.Collection(mongo.CollectionNameUsers).FindOne(ctx, bson.M{"_id": b.Entitlement.ID}).Decode(&user); err != nil {
```

Comment:
Is this valid? Can you decode into a null pointer?

Reply:
Wait it is not, this will cause a panic, you have a pointer and you are taking the pointer of said pointer 

**User is not *User and also the underlying value is nil.